packages/transaction-controller/src/helpers/MultichainHelper.ts

```diff
@@ -0,0 +1,401 @@
+    );
+    networkClientIdsToAdd.forEach((id) => {
+      this.#startTrackingByNetworkClientId(id);
+    });
+  };
+
+  #stopTrackingByNetworkClientId(networkClientId: NetworkClientId) {
+    const trackers = this.#trackingMap.get(networkClientId);
+    if (trackers) {
+      trackers.pendingTransactionTracker.stop();
+      this.#removePendingTransactionTrackerListeners(
+        trackers.pendingTransactionTracker,
+      );
+      trackers.incomingTransactionHelper.stop();
+      this.#removeIncomingTransactionHelperListeners(
+        trackers.incomingTransactionHelper,
+      );
+      this.#trackingMap.delete(networkClientId);
+    }
+  }
+
+  stopAllTracking() {
+    if (!this.#isMultichainEnabled) {
+      return;
+    }
+    for (const [networkClientId] of this.#trackingMap) {
+      this.#stopTrackingByNetworkClientId(networkClientId);
+    }
+  }
+
+  #startTrackingByNetworkClientId(networkClientId: NetworkClientId) {
+    let trackers = this.#trackingMap.get(networkClientId);
+    if (trackers) {
+      return;
+    }
+
+    const networkClient = this.#getNetworkClientById(networkClientId);
+    const { chainId } = networkClient.configuration;
+
+    let etherscanRemoteTransactionSource =
+      this.etherscanRemoteTransactionSourcesMap.get(chainId);
+    if (!etherscanRemoteTransactionSource) {
+      etherscanRemoteTransactionSource = new EtherscanRemoteTransactionSource({
+        includeTokenTransfers:
+          this.#incomingTransactionOptions.includeTokenTransfers,
+      });
+      this.etherscanRemoteTransactionSourcesMap.set(
+        chainId,
+        etherscanRemoteTransactionSource,
+      );
+    }
+
+    trackers = this.#createTrackersForNetworkClient(
+      networkClient,
+      etherscanRemoteTransactionSource,
+    );
+
+    this.#trackingMap.set(networkClientId, trackers);
+  }
+
+  checkForPendingTransactionAndStartPolling = () => {
+    if (!this.#isMultichainEnabled) {
+      return;
+    }
+    for (const [, trackers] of this.#trackingMap) {
+      trackers.pendingTransactionTracker.startIfPendingTransactions();
+    }
+  };
+
+  #refreshEtherscanRemoteTransactionSources = (
+    networkClients: NetworkClientRegistry,
+  ) => {
+    // this will be prettier when we have consolidated network clients with a single chainId:
+    // check if there are still other network clients using the same chainId
+    // if not remove the etherscanRemoteTransaction source from the map
+    const chainIdsInRegistry = new Set();
+    Object.values(networkClients).forEach((networkClient) =>
+      chainIdsInRegistry.add(networkClient.configuration.chainId),
+    );
+    const existingChainIds = Array.from(
+      this.etherscanRemoteTransactionSourcesMap.keys(),
+    );
+    const chainIdsToRemove = existingChainIds.filter(
+      (chainId) => !chainIdsInRegistry.has(chainId),
+    );
+
+    chainIdsToRemove.forEach((chainId) => {
+      this.etherscanRemoteTransactionSourcesMap.delete(chainId);
+    });
+  };
+}
```

Comment:
The fact we've managed to encapsulate 400 lines of logic into here is awesome and feels like excellent justification that this helper is indeed warranted since is illustrates the volume of logic surrounding the concept of tracking / multichain that is now managed in isolation ❤️